library_information_validator.py

```diff
@@ -0,0 +1,318 @@
+        project_id_user = re.findall('P[0-9]+', plate_id)[0]
+        return(project_id_user)
+
+    def projectName_U(self):
+        access_sample_info_sheet = self.getAccessUserSheet()
+        project_name_user = re.split('-', access_sample_info_sheet[ProjectSheet.PROJECT_NAME_USER_SHEET].value)[0]
+        return(project_name_user)
+
+    def getSamples(self):
+        """ identifies the all rows containing a sample name, discards rows without entry.
+        Rows containing whitespace only trigger a warning and are discarded for subsequent
+        tests """
+        access_sample_info_sheet = self.getAccessUserSheet()
+        real = 1
+        cellID_withSample =list()
+        cellID_noSample =list()
+        for i in range(ProjectSheet.FIRST_LINE, ProjectSheet.FIRST_LINE+96):
+            cell_id = "{col}{row_itter}".format(col=ProjectSheet.SAMPLE_NAME_COL,row_itter=i)
+            cell_value = str(access_sample_info_sheet[cell_id].value)
+            if(cell_value.isspace()):
+                logger.warning(
+                    'Cell {} contains empty spaces only. Remove content.'.format(cell_id)
+                   )
+                global WARNINGS
+                WARNINGS = WARNINGS+1
+            elif(access_sample_info_sheet[cell_id].value != None):
+                cellID_withSample.append(i)
+            else:
+                cellID_noSample.append(cell_id)# TODO check here that these rows do really not contain information
+        return(cellID_withSample)
+
+    def ProjectInfo(self, username_couchDB, password_couchDB):
+        url_string = "http://"+username_couchDB+":"+password_couchDB+"@tools-dev.scilifelab.se:5984"
+        connection = couchdb.Server(url=url_string)
+        db = connection["projects"]
+    # check the existence of the project number in couchDB
+        project_id_found = db.view("project/project_id", key=self.projectID())
+        prow=project_id_found.rows
+        # Project not found
+        if len(prow) == 0:
+            logger.error(
+            'Project not found, please check your entry for the PlateID, it should have the format'
+            'PxxxxxPx, where x are numbers. If your Plate ID is correct, contact your project coordinator.'
+            )
+            quit()
+        # more than one project found
+        elif len(prow) > 1:
+            logger.error(
+            'Project ID not unique, please check your entry for the PlateID, it should have the format'
+            'PxxxxxPx, where x are numbers. If your Plate ID is correct, contact your project coordinator.'
+            )
+            quit()
+        else:
+        # puts the Document of the identified project in a new variable "pdoc"
+            pdoc = db.get(prow[0].id)
+            return(pdoc)
+
+    def requirements(self):
+        wb_sample_recommendations = load_workbook(ProjectSheet.sample_recommendation_sheet, read_only=True, data_only=True)
+        ws_recommendations = wb_sample_recommendations['Numeric']
```

Comment:
Same comment as above on getAccessUserSheet to make the code a bit more efficient.